### PR TITLE
perf: upgrade @react-navigation/drawer to v6 and enable freeze

### PR DESCRIPTION
**Pull request by MuckT**
### Description

Updates to [@react-navigation/drawer](https://www.npmjs.com/package/@react-navigation/drawer?activeTab=versions) to 6.5.0. Enables freeze on all screens except Wallet Home: [Read more about React-Freeze](https://github.com/software-mansion/react-freeze). This significantly improves performance by avoiding rerenders on screens that are not visible to the user at a given moment.

### Other changes

Patch for react-native-screens to allow us to continued use of the QR scanner without a refactor.
Moved reset modal into the `BackupPhrase.tsx` due to difficulties with pin entry promise and subsequent navigation.

#### Video of Reset Modal Changes

https://user-images.githubusercontent.com/26950305/199921860-b9171195-8f5d-445b-b7ac-d19c0245e4ce.mov

### Tested

Tested locally on iOS 15.5 iPhone 11 and Android SDK 21 and 29.

### How others should test

- Navigate through the app and ensure that all pages load. 
- Navigate to the QR scanner and back and ensure that pages load.
- Test navigation to and back from all drawer items.

### Related issues

N/A

### Backwards compatibility

Yes

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -73,8 +73,8 @@
     "@celo/wallet-rpc": "~1.2.0",
     "@coinbase/cbpay-js": "^1.0.2",
     "@crowdin/ota-client": "^0.4.0",
-    "@fiatconnect/fiatconnect-sdk": "^0.4.0",
-    "@fiatconnect/fiatconnect-types": "^8.0.0",
+    "@fiatconnect/fiatconnect-sdk": "^0.5.0",
+    "@fiatconnect/fiatconnect-types": "^10.1.0",
```
Comment (kathaypacific) [inline package.json:77]: is this change necessary for "upgrade @react-navigation/drawer to v6 and enable freeze"? if not can we please remove related changes from this PR and open a separate one?

File: package.json [hunk 2]
```diff
@@ -62,7 +62,7 @@
     ]
   },
   "dependencies": {
-    "@badrap/result": "~0.2.12",
+    "@badrap/result": "~0.2.13",
```
Comment (kathaypacific) [inline package.json:65]: same with this, is this directly related to this PR?

**Review by kathaypacific — COMMENTED** [review]
hey the changes on this PR is currently +1840 and -1930 lines, i see there are several things going on in this PR (fiatconnect?) so can we pull it apart to make it more focused and easy to review?

**Comment by MuckT** [comment]
@kathaypacific this PR is so large because it's branched off #3004 after merging that and rebasing it should be much easier to review.

**Comment by kathaypacific** [comment]
> @kathaypacific this PR is so large because it's branched off #3004 after merging that and rebasing it should be much easier to review.

hey @MuckT i did change the base branch of this PR in hopes that the changes would get smaller, and it did (it started off way bigger) but i still think that if there are multiple things unrelated going on in this PR that we should break it apart. do the fiatconnect changes need to be made to enable the drawer v6?

**Comment by MuckT** [comment]
Setting to draft until #3004 is merged.

**Comment by MuckT** [comment]
Closing and will reopen a new PR once #3004 is merged.
